docs/ConfigurationGuide.md

```diff
@@ -90,3 +90,47 @@ Several functions in VC-AuthN can be tweaked by using the following environment
 
 - `include_v1_attributes`: defaults to `false`, switch to `true` if root-level claims as presented in VC-AuthN v1 are still required for the proof-request.
+
+### Proof Substitution Variables
+
+Proof Request configurations (as described [here](README.md#data-model)) are pre-set records stored in the VC-AuthN database comprising of the details to make the proof request from.
+At runtime when the user is directed to the proof challenge, the appropriate configuration is fetched and the proof request built from that.  
+In certain use cases you may want the proof request to have a specific value in the proof (probably in a requested predicate) generated at that moment rather than preset. 
+
+An example could be using today's date, a "right now" timestamp, or an age-check (IE today's date minus 19 years).  
+To accomodate this, VC-AuthN **proof substitution variables** can be used as placeholders in the configurations. They can just be added as string in the configuration with a `$` prefix.
+There are a handful of built-in options:
+
+| Substitution Variable     | Details                                                                                   |
+| ------------------------- | ----------------------------------------------------------------------------------------- |
+| $now                      | Inserts the current timestamp in seconds since the epoch as an int                        |
+| $today_int                | Inserts Today's date in YYYYMMDD format as an int                                         |
+| $tomorrow_int             | Inserts Tomorrow's date in YYYYMMDD format as an int                                      |
+| $threshold_years_X        | Supply a number for X. Inserts today's date minus X years in YYYYMMDD format as an int    |
+
+`$threshold_years_X` is an example of a 'dynamic' substitution variable where part of the variable name can act as a parameter
+
+So in a proof request config a requested predicate could have
+```
+          "p_value": "$today_int",
+          "p_type": ">"
+```
+and at runtime when the user navigates to the QR code page, the proof would include something like
+```
+          "p_value": 20240927,
+          "p_type": ">"
+```
+
+
+See the `oidc-controller\api\verificationConfigs\variableSubstitutions.py` file for implementations.
+
+#### Customizing variables
+
+In `oidc-controller\api\verificationConfigs\variableSubstitutions.py` there are the built-in variables above.
+For an advanced use case, if you require further customization, it could be possible to just replace that `variableSubstitutions.py` file 
+in a VC-AuthN implementation and the newly introduced variables would be run if they are included in a proof request configuration.
+
+For regular variables they can be added to the `static_map`, mapping your variable name to a function doing the operation.  
+For "dynamic" ones alter `__contains__` and `__getitem__` to use a regex to parse and extract what is needed.
```

Comment:
This got me thinking: what about being able to specify a file that `adds` or `replaces` variable substitutions to the `static_map`, rather than replacing the file?

i.e.: default variable substitutions are always available, but if a separate file path is specified we take it's function members and add them to the map so they can be used as well. Not for this PR, but dropping the comment here as it is in context.

Reply:
Yeah we could probably have it look for the presence of another file and then import functions. The current model would need an operator to overwrite the existing file (and could end up with conflicts if the default file gets changed etc).

Reply:
I'll log something for us to look into - that was exactly my thinking.